lib/rules/object-curly-spacing.js

```diff
@@ -121,9 +121,10 @@ module.exports = function(context) {
      * @returns {void}
      */
     function validateBraceSpacing(node, first, second, penultimate, last) {
+        var penultimateType = sourceCode.getNodeByRangeIndex(penultimate.start).type;
```

Comment:
So you're suggesting both my "calculate at the last moment" thing and memoizing `getNodeByRangeIndex` method? Or are you just suggesting caching inside the rule?